### PR TITLE
SOF-6827: 3pse schemas

### PR DESCRIPTION
**Pull request by k0stik**

### CODE REVIEW

File: schema/3pse/file/applications/espresso/7.2/pw.x/system.json [hunk 1]
```diff
@@ -0,0 +1,638 @@
...
+            "type": "number",
+            "default": 0,
+            "description": "Overlap threshold over which the exchange integral over a pair of localized orbitals is included in the evaluation of EXX operator. Any value greater than 0.0 triggers the SCDM localization and the evaluation on EXX using the localized orbitals. Very small value of the threshold should yield the same result as the default EXX evaluation"
+        },
+        "Hubbard_occ": {
+            "type": "array",
+            "items": {
+                "type": "array",
+                "items": {
+                    "type": "number"
```
Comment (pranabdas) [inline schema/3pse/file/applications/espresso/7.2/pw.x/system.json]: Hi @k0stik, just a suggestion, here we could enforce stricter tuple validation:
```json
"Hubbard_occ": {
    "type": "array",
    "items": {
        "type": "array",
        "items": [
            { "type": "integer", "minimum": 1 },
            { "type": "integer", "minimum": 1, "maximum": 3 },
            { "type": "number" }
        ],
        "maxItems": 3,
        "minItems": 3
    }
},
```
First two items in the array are indices of atomic species and Hubbard manifold. Both are non-negative integers and there could be up to 3 Hubbard manifolds. Note that if you accept this suggestion, we need to update the example data:
```
"Hubbard_occ": [
            [1, 1, 0.0],
            [1, 2, 0.02]
        ]
```

File: schema/3pse/file/applications/espresso/7.2/pw.x/system.json [hunk 2]
```diff
@@ -0,0 +1,638 @@
...
+            "type": "number",
+            "default": 0,
+            "description": "starting charge on atomic type 'i', to create starting potential with startingpot = 'atomic'."
+        },
+        "tot_magnetization": {
+            "type": "number",
+            "default": -10000,
+            "description": "Total majority spin charge - minority spin charge. Used to impose a specific total electronic magnetization. If unspecified then tot_magnetization variable is ignored and the amount of electronic magnetization is determined during the self-consistent cycle."
+        },
+        "starting_magnetization": {
```
Comment (pranabdas) [inline schema/3pse/file/applications/espresso/7.2/pw.x/system.json:77]: `starting_magnetization` can have values in the range [-1, 1]. We may specify `minimum` and `maximum`.

File: schema/3pse/file/applications/espresso/7.2/pw.x/atomic_species.json [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+    "type": "object",
+    "additionalProperties": false,
+    "properties": {
+        "values": {
+            "type": "array",
+            "items": {
+                "type": "object",
+                "additionalProperties": false,
+                "properties": {
+                    "x": {
```
Comment (timurbazhirov) [inline schema/3pse/file/applications/espresso/7.2/pw.x/atomic_species.json]: Should be capital "X" I think

File: schema/3pse/file/applications/espresso/7.2/pw.x/atomic_positions.json [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+        },
+        "values": {
+            "type": "array",
+            "items": {
+                "type": "object",
+                "additionalProperties": false,
+                "properties": {
+                    "X": {
+                        "type": "string",
+                        "description": "label of the atom as specified in ATOMIC_SPECIE"
```
Comment (timurbazhirov) [inline schema/3pse/file/applications/espresso/7.2/pw.x/atomic_positions.json]: SPECIES

File: schema/3pse/file/applications/espresso/7.2/pw.x/atomic_positions.json [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+                    },
+                    "z": {
+                        "type": "number",
+                        "description": "atomic positions"
+                    },
+                    "if_pos(1)": {
+                        "type": "number",
+                        "default": 1,
+                        "minimum": 0,
+                        "maximum": 1,
```
Comment (timurbazhirov) [inline schema/3pse/file/applications/espresso/7.2/pw.x/atomic_positions.json]: 1. In general, there's a lot of duplication in how we define each key, we should identify reusable schema bits and isolate them, then use here. This one could be something like `integer_one_or_zero`
2. We should restrict the values here to only 0 or 1, so put "integer" as a type

File: schema/3pse/file/applications/espresso/7.2/pw.x/control.json [hunk 6]
```diff
@@ -0,0 +1,179 @@
...
+            "type": "number",
+            "description": "For Berry phase calculation: direction of the k-point strings in reciprocal space. Allowed values: 1, 2, 3 1=first, 2=second, 3=third reciprocal lattice vector For calculations with finite electric fields (lelfield==.true.) \"gdir\" is the direction of the field."
+        },
+        "nppstr": {
+            "type": "number",
+            "description": "For Berry phase calculation: number of k-points to be calculated along each symmetry-reduced string. The same for calculation with finite electric fields (lelfield==.true.)."
+        },
+        "gate": {
+            "type": "boolean",
+            "default": false
```
Comment (timurbazhirov) [inline schema/3pse/file/applications/espresso/7.2/pw.x/control.json]: There is description for both this and `trism` fields in https://www.quantum-espresso.org/Doc/INPUT_PW.html.

**Review by timurbazhirov — APPROVED** [review]

File: schema/core/primitive/integer_one_or_zero.json [hunk 7]
```diff
@@ -0,0 +1,8 @@
+{
+    "$id": "core/primitive/integer-one-or-zero",
+    "$schema": "http://json-schema.org/draft-04/schema#",
+    "title": "integer one or zero",
+    "type": "number",
```
Comment (timurbazhirov) [inline schema/core/primitive/integer_one_or_zero.json]: Type "integer"?

**Review by azech-hqs — APPROVED** [review]
needs to be update/rebased to current `dev`, otherwise looks good 👍

**Review by timurbazhirov — APPROVED** [review]

**Pull request merged**